packages/foundry-sdk-generator/src/generate/betaClient/minifyBundle.ts

```diff
@@ -231,6 +357,28 @@ export class ProjectMinifier {
           importSourceFile,
         );
       }
+
+      // get the dynamic import statements
+      sourceFile.getDescendantsOfKind(SyntaxKind.ImportType).forEach((node) => {
+        const moduleSpecifier = node.getArgument().getText().slice(1, -1);
+        const importSourceFile = getModuleSourceFile(
+          this.project,
+          moduleSpecifier,
+        );
+        const qualifier = node.getQualifier();
+        const qualifierSymbol = qualifier?.getSymbol();
+
+        if (
+          qualifier && qualifierSymbol && importSourceFile
+          && identifiersToResolve.has(qualifierSymbol)
+        ) {
+          this.pushNextVisit(
```

Comment:
This just ensures that if a type has `const foo: import("./bar").Baz;` that we ensure that `Baz` is kept in `"./bar"`